projects/driver_controls/src/mech_brake.c

```diff
@@ -0,0 +1,78 @@
+
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <stdbool.h>
+#include "ads1015.h"
+#include "delay.h"
+#include "event_arbiter.h"
+#include "event_queue.h"
+#include "gpio_it.h"
+
+#include "input_event.h"
+#include "log.h"
+#include "mech_brake.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "wait.h"
+
+// takes in LSB as input and converts it to percentage using a linear relationship
+static int16_t lsb_to_percentage_converter(MechBrakeStorage *storage) {
+  int16_t percentage;
+
+  if (storage->calibration_data->zero_value > storage->calibration_data->hundred_value) {
+    percentage =
+        ((storage->settings.min_allowed_range *
+          (storage->reading - storage->calibration_data->hundred_value)) /
+         (storage->calibration_data->hundred_value - storage->calibration_data->zero_value)) +
+        storage->settings.max_allowed_range;
+  } else {
+    percentage = (storage->settings.max_allowed_range *
+                  (storage->reading - storage->calibration_data->zero_value)) /
+                 (storage->calibration_data->hundred_value - storage->calibration_data->zero_value);
+  }
+
+  if (percentage < storage->settings.min_allowed_range) {
+    percentage = storage->settings.min_allowed_range;
+  } else if (percentage > storage->settings.max_allowed_range) {
+    percentage = storage->settings.max_allowed_range;
+  }
+
+  // add code to see if the percentage is greater than max allowed value
+  if (percentage > storage->settings.percentage_threshold) {
+    uint16_t percentage_data = (uint16_t)percentage;
```

Comment:
Why are you declaring the same variable twice? What's the point of this extra variable in the first place?

Reply:
the percentage variable is initialized as an integer but I need to cast it into an unsigned integer in order to use it in the event_raise function